libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -307,13 +307,15 @@ internal class Kotlin2JsSourceSetProcessor(
 
     private fun registerCleanSourceMapTask() {
         val taskName = kotlinCompilation.composeName("clean", "sourceMap")
-        val registerTask = registerTask(project, taskName, Delete::class.java) {
+        project.registerTask<Delete>(taskName) {
             it.onlyIf { kotlinTask.get().kotlinOptions.sourceMap }
```

Comment:
I am wondering here when the `onlyIf` statement is gonna be evaluated as calling `get()` will realize the task.
TBH, it would be nice to have someone from Gradle having a look at it.